_posts/2022-02-14-datafusion-7.0.0.md

```diff
@@ -0,0 +1,166 @@
+     3  Remzi Yang
+     3  Yang
+     3  ic4y
+     3  Daniël Heres
+     2  Andy Grove
+     2  Raphael Taylor-Davies
+     2  Jason Tianyi Wang
+     2  Dan Harris
+     2  Sergey Melnychuk
+     1  Nitish Tiwari
+     1  Dom
+     1  Eduard Karacharov
+     1  Javier Goday
+     1  Boaz
+     1  Marko Mikulicic
+     1  Max Burke
+     1  Carol (Nichols || Goulding)
+     1  Phillip Cloud
+     1  Rich
+     1  Toby Hede
+     1  Will Jones
+     1  r.4ntix
+     1  rdettai
+```
+
+The release notes below are not exhaustive and only expose selected highlights of the release. Many other bug fixes
+and improvements have been made: we refer you to the complete
+[changelog](https://github.com/apache/arrow-datafusion/blob/7.0.0/datafusion/CHANGELOG.md).
+
+# Summary
+
+There have been significant improvements across the board since the 6.0 release which are summarized below.
+
+- DataFusion Crate
+  - The DataFusion crate is in the process of being split into multiple crates in order to decrease compilation times and improve the development experience. To start, datafusion-common (the core DataFusion components) and datafusion-expr (DataFusion expressions, functions, and operators) will be split out.  There will be additional splits after the 7.0 release.
+- Performance Improvements and Optimizations
+  - Arrow’s dyn scalar kernels are now used which enable more efficient operations on DictionaryArrays [#1685](https://github.com/apache/arrow-datafusion/pull/1685)
+  - Switch from std::sync::Mutex to parking_lot::Mutex [#1720](https://github.com/apache/arrow-datafusion/pull/1720)
+- New Features
+  - Better support for limiting resource usage
+    - MemoryMananger and DiskManager [#1526](https://github.com/apache/arrow-datafusion/pull/1526)
+    - Out of core sort [#1526](https://github.com/apache/arrow-datafusion/pull/1526)
+    - New metrics
+      - `Gauge` and `CurrentMemoryUsage` [#1682](https://github.com/apache/arrow-datafusion/pull/1682)
+      - `Spill_count` and `spilled_bytes` [#1641](https://github.com/apache/arrow-datafusion/pull/1641)
+  - New math functions
+    - `Approx_quantile` [#1529](https://github.com/apache/arrow-datafusion/pull/1539)
+    - `stddev` and `variance` (sample and population) [#1525](https://github.com/apache/arrow-datafusion/pull/1525)
+    - `corr` [#1561](https://github.com/apache/arrow-datafusion/pull/1561)
+  - Support decimal type [#1394](https://github.com/apache/arrow-datafusion/pull/1394)[#1407](https://github.com/apache/arrow-datafusion/pull/1407)[#1408](https://github.com/apache/arrow-datafusion/pull/1408)[#1431](https://github.com/apache/arrow-datafusion/pull/1431)[#1483](https://github.com/apache/arrow-datafusion/pull/1483)[#1554](https://github.com/apache/arrow-datafusion/pull/1554)[#1640](https://github.com/apache/arrow-datafusion/pull/1640)
+  - Support for evolved schemas [#1622](https://github.com/apache/arrow-datafusion/pull/1622)[#1709](https://github.com/apache/arrow-datafusion/pull/1709)
```

Comment:
```suggestion
  - Support for reading parquet files with evolved schemas [#1622](https://github.com/apache/arrow-datafusion/pull/1622)[#1709](https://github.com/apache/arrow-datafusion/pull/1709)
```

Reply:
I had thought #1709 helped generalize schema adaptors to more than just Parquet files.  But i didnt get to dig in too much to that so maybe i misunderstood

Reply:
i think @thinkharderdev concluded the schema merging already worked for CSV and Json files, but I may also misunderstand

Reply:
That's right. Merging already worked for everything except Parquet